samtranslator/swagger/swagger.py

```diff
@@ -852,6 +852,8 @@ def add_resource_policy(self, resource_policy, path, api_id, stage):
         """
         if resource_policy is None:
             return
+        if not isinstance(resource_policy, dict):
+            raise InvalidDocumentException([InvalidTemplateException("Resource Policy is not a valid dictionary.")])
```

Comment:
Curious, what is benefit of wrapping an array of exceptions in  another exception?

Reply:
This is how it is meant to be used
```
    Attributes:
        message -- explanation of the error
        causes -- list of errors which caused this document to be invalid
    """
```

Reply:
No idea why. This is what InvalidDocumentException needs. I saw other examples and this is what they did. I tried other ways but none works.

Reply:
afaict, InvalidDocumentException is meta exception, which counts all sub exceptions and then produces a single exception message combining them all.